File: backend/anavis/src/main/java/progetto/anavis/model/Questionario.java
package progetto.anavis.model;

import java.util.UUID;

import javax.persistence.Entity;
import javax.persistence.Id;
import javax.validation.constraints.NotNull;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Questa classe definisce l'oggetto questionario che comprende un insieme di
 * domande alle quali il donatore deve rispondere per fornire informazioni
 * riguardo il suo stato di salute prima di effettuare la donazione.
 * 
 * @author Lucia e Luca
 *
 */

@Entity
public class Questionario {

	@Id
	private UUID id;
	private String motiviRicovero, qualiAllergie;
	@NotNull
	private boolean buonaSalute, ricoveratoOspedale, condizioniSaluteRecenti, allergie, perditaPeso;

	public Questionario() {

	}

	public Questionario(@JsonProperty("id") UUID id, @JsonProperty("salute attuale") boolean saluteAttuale,
			@JsonProperty("ricovero") boolean ricoverato, @JsonProperty("motivi ricovero") String motiviRicovero,
			@JsonProperty("salute recente") boolean recentiCondizioni, @JsonProperty("allergie") boolean allergie,
			@JsonProperty("elenco allergie") String elencoAllergie,
			@JsonProperty("perdita di peso") boolean perditaDiPeso) {
		this.id = id;
		this.buonaSalute = saluteAttuale;
		this.ricoveratoOspedale = ricoverato;
		this.motiviRicovero = motiviRicovero;
		this.condizioniSaluteRecenti = recentiCondizioni;
		this.allergie = allergie;
		this.qualiAllergie = elencoAllergie;
		this.perditaPeso = perditaDiPeso;
	}

	public UUID getId() {
		return id;
	}

	public void setId(UUID id) {
		this.id = id;
	}

	public String getMotiviRicovero() {
		return motiviRicovero;
	}

	public void setMotiviRicovero(String motiviRicovero) {
		this.motiviRicovero = motiviRicovero;
	}

	public String getQualiAllergie() {
		return qualiAllergie;
	}

	public void setQualiAllergie(String qualiAllergie) {
		this.qualiAllergie = qualiAllergie;
	}

	public boolean isBuonaSalute() {
		return buonaSalute;
	}

	public void setBuonaSalute(boolean buonaSalute) {
		this.buonaSalute = buonaSalute;
	}

	public boolean isRicoveratoOspedale() {
		return ricoveratoOspedale;
	}

	public void setRicoveratoOspedale(boolean ricoveratoOspedale) {
		this.ricoveratoOspedale = ricoveratoOspedale;
	}

	public boolean isCondizioniSaluteRecenti() {
		return condizioniSaluteRecenti;
	}

	public void setCondizioniSaluteRecenti(boolean condizioniSaluteRecenti) {
		this.condizioniSaluteRecenti = condizioniSaluteRecenti;
	}

	public boolean isAllergie() {
		return allergie;
	}

	public void setAllergie(boolean allergie) {
		this.allergie = allergie;
	}

	public boolean isPerditaPeso() {
		return perditaPeso;
	}

	public void setPerditaPeso(boolean perditaPeso) {
		this.perditaPeso = perditaPeso;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null || getClass() != obj.getClass())
			return false;
		Questionario questionario = (Questionario) obj;
		return this.id.equals(questionario.id);
	}

	@Override
	public String toString() {
		return "Id: " + this.id;
	}

}

File: frontend/an_avis/lib/models/questionario.dart
import "package:flutter/material.dart";

class QuestionarioProvider extends ChangeNotifier {
  bool _buonaSalute = false;
  bool _ricoveratoOspedale = false;
  bool _condizioniSaluteRecenti = false;
  bool _allergie = false;
  bool _perditaPeso = false;
  String _motiviRicovero;
  String _qualiAllergie;

  void setBuonaSalute(bool value) {
    _buonaSalute = value;
  }

  void setRicoveratoOspedale(bool value) {
    _ricoveratoOspedale = value;
  }

  void setCondizioniSaluteRecenti(bool value) {
    _condizioniSaluteRecenti = value;
  }

  void setAllergie(bool value) {
    _allergie = value;
  }

  void setPerditaPeso(bool value) {
    _perditaPeso = value;
  }

  void setMotiviRicovero(String motivo) {
    _motiviRicovero = motivo;
  }

  void setQualiAllergie(String allergie) {
    _qualiAllergie = allergie;
  }

  bool getBuonaSalute() {
    return _buonaSalute;
  }

  bool getRicoveroOspedale() {
    return _ricoveratoOspedale;
  }

  bool getCondizioniSaluteRecenti() {
    return _condizioniSaluteRecenti;
  }

  bool getAllergie() {
    return _allergie;
  }

  bool getPerditaPeso() {
    return _perditaPeso;
  }

  String getMotiviRicovero() {
    return _motiviRicovero;
  }

  String getQualiAllergie() {
    return _qualiAllergie;
  }
}
